### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by lucasdemarchi**

### CODE REVIEW

File: libraries/AP_HAL_Linux/GPIO_RPI.h [hunk 1]
```diff
@@ -6,69 +6,42 @@
...
 #define RPI_GPIO_29   29   // Pin 4
 #define RPI_GPIO_30   30   // Pin 5
 #define RPI_GPIO_31   31   // Pin 6
 
 class Linux::GPIO_RPI : public AP_HAL::GPIO {
-private:
-    int  mem_fd;
-    void *gpio_map;
-    volatile uint32_t *gpio;
-
```
Comment (guludo) [inline libraries/AP_HAL_Linux/GPIO_RPI.h]: "AP_HAL_Linux: GPIO: reduce macro scopes"

You're doing more than that.

Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/GPIO_RPI.h]: just reword the commit to reduce macro/variable scopes

File: libraries/AP_HAL_Linux/GPIO_RPI.cpp [hunk 2]
```diff
@@ -20,23 +20,39 @@
...
+}
 
 void GPIO_RPI::init()
 {
     int rpi_version = UtilRPI::from(hal.util)->get_rpi_version();
     uint32_t gpio_address = rpi_version == 1 ? GPIO_BASE(BCM2708_PERI_BASE)   : GPIO_BASE(BCM2709_PERI_BASE);
-    // open /dev/mem
-    if ((mem_fd = open("/dev/mem", O_RDWR|O_SYNC) ) < 0) {
+
+    int mem_fd = open("/dev/mem", O_RDWR|O_SYNC);
```
Comment (guludo) [inline libraries/AP_HAL_Linux/GPIO_RPI.cpp]: Hm... I know that the original implementation wasn't doing that. But I think `mem_fd` should be closed when this function exits.

Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/GPIO_RPI.cpp]: I think you misread the commit. We already close it and there's even a comment: `No need to keep mem_fd open after mmap`

**Comment by lucasdemarchi** [comment]
I reworded 1 commit message as requested and applied.

**Pull request closed without merging**